### PR TITLE
[fusion] Add ConvTranspose+BN fusion support

### PR DESCRIPTION
**Pull request by jonmorton**
Summary: Add support for fusing ConvTranpose{1,2,3}d with BatchNorm{1,2,3}d. This re-uses the existing fusion logic but adds a "transpose" flag to the fusing function which when enabled will use the appropriate reshape for ConTranspose's transposed weights.

Test Plan: `buck test mode/dev //caffe2/test:quantization -- -r quantization.eager.test_fusion.TestFusion`

Differential Revision: D33074405

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
looks great, thanks for working on this!

**Comment by jerryzh168** [comment]
closes: https://github.com/pytorch/pytorch/issues/69797
